### PR TITLE
Refactor ZoomLevel to re-use GeographicLevel and hide pills in map view 

### PR DESCRIPTION
**Pull request by justinebreuch**
## Description

Addresses: [Hide pills on scorecard page and disable selection in side panel](https://tables.area120.google.com/u/1/workspace/bxQE2CPzwu_b4sJEkotkUM/table/bX_kMNBW8LA35n6l5Hz_co/view/auSM-s1cO9l3pAsYHmOk1E)

1. Hides the pills behind a feature called enableViewToggling in ScorecardSidePanel. Checked to make sure they work if turned on. 
2. Disables view selection behind enableSelection in GeoIdSelection
3. Removes "zoomLevel" from mapState in favor of geographicView that reuses GeographicLevel already referenced througout the UI. This caused issue because we had zoom and zoomLevel, different components were listening to different things causes discrepancies.

### Changed

ZoomLevel --> GeographicLevel 

### Removed

ZoomLevel

## Testing and Reviewing

Walked through all UI flows in Map and Scorecard. Logged new bugs found (also in dev).

### CODE REVIEW

File: client/src/model/store_plugin.ts [hunk 1]
```diff
@@ -15,7 +15,6 @@ const createRedux = (store: EnhancedStore) => {
 
       store.subscribe(() => {
         rootStore.state = store.getState();
-        console.log(`The state has been updated ${JSON.stringify(rootStore.state)}`);
```
Comment (justinebreuch) [inline client/src/model/store_plugin.ts:18]: This is very noisy

File: client/src/model/data_layer.ts [hunk 2]
```diff
@@ -121,15 +121,17 @@ enum MapLayer {
 }
 
 /**
- * Zoom levels for each demographic layer.
+ * Zoom thresholds for each geographic level.
```
Comment (beekley) [inline client/src/model/data_layer.ts:124]: Does the enum applies if it's greater or less than the threshold? i.e. does `State` apply for 0 to 4, or 4 to 5?

(then could you add whatever the answer is to the comment here)

File: client/src/components/scorecard_page/PredictionPanel.vue [hunk 3]
```diff
@@ -49,17 +51,21 @@ export default defineComponent({
...
-      if (this.showParcelPrediction)
+      }
+      if (this.showParcelPrediction) {
         return (
           this.formatPredictionAsLikelihood(this.publicLeadPercent) ??
           this.ScorecardSummaryMessages.NOT_ENOUGH_DATA_AVAILABLE
         );
-      if (this.showNoPrediction) return this.ScorecardSummaryMessages.NOT_ENOUGH_DATA_AVAILABLE;
+      }
+      if (this.showNoPrediction) {
```
Comment (beekley) [inline client/src/components/scorecard_page/PredictionPanel.vue:66]: (this is outside the scope of this PR) I don't love the negative for this property, when the others are positive. What's the difference between `showNoPrediction` and `!showParcelPrediction`?

File: client/src/data_layer_configs/population_by_census_block_config.ts [hunk 4]
```diff
@@ -104,7 +104,7 @@ const legendInterpolation = [
 
 const legendInterpolationCounty = [
```
Comment (beekley) [inline client/src/data_layer_configs/population_by_census_block_config.ts:105]: nit: Should this variable name change to say zip as well?

File: client/src/model/store_plugin.ts (hunk 1 above)
Comment (beekley) [reply client/src/model/store_plugin.ts:18]: +1. Though I'm up to still log some stuff that's useful.

**Review by beekley — APPROVED** [review]

File: client/src/model/data_layer.ts (hunk 2 above)
Comment (justinebreuch) [reply client/src/model/data_layer.ts:124]: Added a comment to clarify thanks

File: client/src/data_layer_configs/population_by_census_block_config.ts (hunk 4 above)
Comment (justinebreuch) [reply client/src/data_layer_configs/population_by_census_block_config.ts:105]: Good catch

File: client/src/components/scorecard_page/PredictionPanel.vue (hunk 3 above)
Comment (justinebreuch) [reply client/src/components/scorecard_page/PredictionPanel.vue:66]: showNoPrediction means there is neither parcel nor water system prediction
